aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -0,0 +1,501 @@
+    {
+    }
+}
+```
+
+Creating the `DomainRestricted` policy might involve:
+
+* In *Startup.cs*, adding the new policy.
+* Provide the custom `DomainRestrictedRequirement` requirement as a parameter.
+* Registering `DomainRestricted` with the authorization middleware.
+
+```csharp
+services
+    .AddAuthorization(options =>
+    {
+        options.AddPolicy("DomainRestricted", policy =>
+        {
+            policy.Requirements.Add(new DomainRestrictedRequirement());
+        });
+    });
+```
+
+SignalR hubs use [Endpoint Routing](xref:fundamentals/routing). SignalR hub connection was previously done explicitly:
+
+```csharp
+app.UseSignalR(routes =>
+{
+    routes.MapHub<ChatHub>("hubs/chat");
+});
+```
+
+In the previous version, developers needed to wire up controllers, Razor pages, and hubs in a variety of different places. Explicit connection results in a series of nearly-identical routing segments:
+
+```csharp
+app.UseSignalR(routes =>
+{
+    routes.MapHub<ChatHub>("hubs/chat");
+});
+
+app.UseRouting(routes =>
+{
+    routes.MapRazorPages();
+});
+```
+
+SignalR 3.0 hubs can be routed via endpoint routing. With endpoint routing, typically all routing can be configured in `UseRouting`:
+
+```csharp
+app.UseRouting(routes =>
+{
+    routes.MapRazorPages();
+    routes.MapHub<ChatHub>("hubs/chat");
+});
+```
+
+ASP.NET Core 3.0 SignalR added:
+
+Client-to-server streaming. With client-to-server streaming, server-side methods can take instances of either an `IAsyncEnumerable<T>` or `ChannelReader<T>`. In the following C# sample, the `UploadStream` method on the Hub will receive a stream of strings from the client:
+
+```csharp
+public async Task UploadStream(IAsyncEnumerable<string> stream)
+{
+    await foreach (var item in stream)
+    {
+        // process content
+    }
+}
+```
+
+.NET client apps can pass either an `IAsyncEnumerable<T>` or `ChannelReader<T>` instance as the `stream` argument of the `UploadStream` Hub method above.
+
+```csharp
+async IAsyncEnumerable<string> clientStreamData()
+{
+    for (var i = 0; i < 5; i++)
+    {
+        var data = await FetchSomeData();
+        yield return data;
+    }
+    //After the for loop has completed and the local function exits the stream completion is sent.
+}
+
+await connection.SendAsync("UploadStream", clientStreamData());
+```
+
+JavaScript client apps use the SignalR `Subject` (or an [RxJS Subject](https://rxjs.dev/api/index/class/Subject)) for the `stream` argument of the `UploadStream` Hub method above.
+
+```javascript
+let subject = new signalR.Subject();
+await connection.send("StartStream", "MyAsciiArtStream", subject);
+```
+
+The JavaScript code could use the `subject.next` method to handle strings as they are captured and ready to be sent to the server.
+
+```javascript
+subject.next("example");
+subject.complete();
+```
+
+Using code like the two preceding snippets, real-time streaming experiences can be created.
+
+## Use the ASP.NET Core shared framework
```

Comment:
How about we move this section and the following section on assemblies removed from the shared framework to the bottom of the article?